### PR TITLE
MINOR: enhance streams system test

### PR DESCRIPTION
**Pull request by ymatsuda**
@guozhangwang 
- add table aggregate to the system test
- actually create change log partition replica

### CODE REVIEW

File: streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java [hunk 1]
```diff
@@ -293,11 +293,9 @@ public StreamsConfig(Map<?, ?> props) {
 
     private void removeStreamsSpecificConfigs(Map<String, Object> props) {
         props.remove(StreamsConfig.JOB_ID_CONFIG);
-        props.remove(StreamsConfig.REPLICATION_FACTOR_CONFIG);
```
Comment (guozhangwang) [inline streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java:296]: Why removing these two?

File: tests/kafkatest/tests/streams_bounce_test.py [hunk 2]
```diff
@@ -58,7 +59,7 @@ def test_bounce(self):
         # enable this after we add change log partition replicas
         #self.kafka.signal_leader("data")
 
-        time.sleep(15);
+        #time.sleep(15);
```
Comment (guozhangwang) [inline tests/kafkatest/tests/streams_bounce_test.py:62]: Is this intentional?

File: streams/src/test/java/org/apache/kafka/streams/smoketest/SmokeTestDriver.java [hunk 3]
```diff
@@ -461,6 +466,42 @@ private static boolean verifyWCnt(Map<String, Long> map) {
...
+                String cnt = Long.toString(max - min + 1L);
+
+                if (expected.containsKey(cnt)) {
+                    expected.put(cnt, expected.get(cnt) + 1L);
+                } else {
+                    expected.put(cnt, 1L);
+                }
+            }
+
+            // check the result
```
Comment (guozhangwang) [inline streams/src/test/java/org/apache/kafka/streams/smoketest/SmokeTestDriver.java:526]: Should we also check if some expected count is missing? I.e. in `expected` but not in `map`.

**Comment by guozhangwang** [comment]
Checkstyle fails in Jenkins: `:streams:checkstyleMain[ant:checkstyle] /home/jenkins/jenkins-slave/workspace/kafka-trunk-git-pr-jdk7/streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopicManager.java:27:8: Unused import - org.apache.kafka.common.utils.Utils.
 FAILED`

File: streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java (hunk 1 above)
Comment (ymatsuda) [inline streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java:296]: StreamPartitionAssignor needs them.

File: streams/src/test/java/org/apache/kafka/streams/smoketest/SmokeTestDriver.java (hunk 3 above)
Comment (ymatsuda) [inline streams/src/test/java/org/apache/kafka/streams/smoketest/SmokeTestDriver.java:526]: good point.

File: tests/kafkatest/tests/streams_bounce_test.py (hunk 2 above)
Comment (ymatsuda) [inline tests/kafkatest/tests/streams_bounce_test.py:62]: Yes.

**Comment by ymatsuda** [comment]
I know. The fix is coming up.

**Comment by ymatsuda** [comment]
Note that now standby tasks are closed before closing the restore consumer in stream thread shutdown. The reason is because removeStandbyTask uses the restore consumer to unassign all partitions.

**Comment by guozhangwang** [comment]
LGTM, merged to trunk.

**Pull request closed without merging**